### PR TITLE
Dev hbbtag heppy745

### PR DESCRIPTION
**Pull request by gkasieczka**
Re-opening PR 60 (#60) against vhbbHeppy74X

### CODE REVIEW

**Comment by arizzi** [comment]
can you please rebase/fix conflicts with #90

**Comment by arizzi** [comment]
I meant #91

**Comment by gkasieczka** [comment]
Should merge cleanly now with PR91 included

**Comment by arizzi** [comment]
any news on the memleak?

**Comment by gkasieczka** [comment]
Ciao Andrea,

a large fraction seems to come from:
    flightDir_0 = ROOT.Vector3DBase(float, ROOT.GlobalTag)()
    flightDir_1 = ROOT.Vector3DBase(float, ROOT.GlobalTag)()

    SV_p4_0 = ROOT.reco.Candidate.LorentzVector()
    SV_p4_1 = ROOT.reco.Candidate.LorentzVector()

Unfortunately calling their destructor explicitely does not solve it.

Cheers,
Gregor

**Comment by arizzi** [comment]
are those needed? I mean why initializing them?

**Comment by arizzi** [comment]
ops didn't mean to close, just comment

**Comment by arizzi** [comment]
it seems to me even commenting those it still leaks... also removing everything using SV_p4 and flightDir

**Comment by gkasieczka** [comment]
So these should be responsible for around 50% of the leak. You are right - in principle we should be able to work without the explicit initializiation in Python. I will check this.

**Comment by arizzi** [comment]
removing the explicit initialization I get to 20KBytes/second from 160KBytes/seconds.
Can you please meanwhile comment out those constructors? we can then chase the remaining leak

File: VHbbAnalysis/Heppy/python/AdditionalBoost.py [hunk 1]
```diff
@@ -1,87 +1,485 @@
...
+
+    # End of loop over vertices
+
+    cont=0
+
+    # GlobalVector is defined DataFormats/GeometryVector/interface/GlobalVector.h
+    # typedef Global3DVector GlobalVector;
+    # typedef Vector3DBase< float, GlobalTag>    Global3DVector;
+    # Since there is no dict for GlobalVector build it directly from the template
+    flightDir_0 = ROOT.Vector3DBase(float, ROOT.GlobalTag)()
```
Comment (arizzi) [inline VHbbAnalysis/Heppy/python/AdditionalBoost.py]: @gpetruc @Dr15Jones do you have any clue why this line (and the next 4 lines), introduce a leak of around 45Kb/event?

**Comment by gkasieczka** [comment]
They're out

**Comment by Dr15Jones** [comment]
@arizzi I got an email question to me pointing to some code but it appears that code is now gone. I believe there was a question about the lifetime of a PyROOT object? For that, you're best off contacting the ROOT team.

**Comment by arizzi** [comment]
@gkasieczka how did you figure that those two lines were the culprit? I mean we still have a leak that make the jobs killed after few hours for too much mem and I'd like to figure where it comes from...

**Comment by arizzi** [comment]
@Dr15Jones I did open an issue in root JIRA, but Wim is complaining that the leaking stuff is not a (Py)ROOT class (Vector3DBase is CMSSW). 

Here the jira ticket link: https://sft.its.cern.ch/jira/browse/ROOT-7484

**Comment by Dr15Jones** [comment]
I added a comment to that Jira thread showing the same problem occurs when using just ROOT's own Vector3D classes.

**Pull request merged**